common/src/main/java/oracle/kubernetes/common/utils/CommonUtils.java

```diff
@@ -33,4 +43,45 @@ public static String toDns1123LegalName(String value) {
     return value.toLowerCase().replace('_', '-');
   }
 
+  /**
+   * Returns a truncated volume name if the name exceeds the max allowed limit of characters for volume name.
+   * @param volumeName volume name
+   * @return truncated volume name if the name exceeds the limit, else return volumeName
+   * @throws NoSuchAlgorithmException Thrown when particular cryptographic algorithm
+   *                                  is not available in the environment.
+   */
+  public static String getLegalVolumeName(String volumeName) throws NoSuchAlgorithmException {
+    return volumeName.length() > (MAX_ALLOWED_VOLUME_NAME_LENGTH)
+        ? getShortName(volumeName)
+        : volumeName;
+  }
+
+
+  private static String getShortName(String resourceName) throws NoSuchAlgorithmException {
+    String volumeSuffix = Optional.ofNullable(getMD5Hash.apply(resourceName)).orElse("");
+    return resourceName.substring(0, MAX_ALLOWED_VOLUME_NAME_LENGTH - volumeSuffix.length()) + volumeSuffix;
+  }
+
+  /**
+   * Gets the MD5 hash of a string.
+   *
+   * @param data input string
+   * @return MD5 hash value of the data, null in case of an exception.
+   */
+  public static String getMD5Hash(String data) throws NoSuchAlgorithmException {
```

Comment:
is this method different from getMD5Hash() in ChecksumUtils?

Reply:
The ChecksumUtils class is deleted in this PR. I moved the method from ChecksumUtils to CommonUtils.

Reply:
nevermind. I see that ChecksumUtils.java was deleted.